airflow/models/taskinstance.py

```diff
@@ -323,6 +323,8 @@ def _run_raw_task(
                 return None
             else:
                 ti.handle_failure(e, test_mode, context, session=session)
+                if ti.state == TaskInstanceState.UP_FOR_RETRY:
+                    ti.dag_run.next_schedulable = ti.next_retry_datetime()
```

Comment:
I'm not sure this is always right - what if another task in the dag has set this to something else as it's schedulable?

Reply:
I will experiment with this

Reply:
I have changed this. It leads to delays in retry for some cases. I'm now using a timer to run a simple and optimized check in the scheduler to determine when to enable scheduling for a dagrun. I used 5s instead of a configuration and I think it's ok.